bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaBridgeHandler.java

```diff
@@ -165,6 +174,14 @@ public synchronized void login() {
 
     }
 
+    private String urlEncode(String text) throws UnsupportedEncodingException {
+        return URLEncoder.encode(text, StandardCharsets.UTF_8.toString());
+    }
+
+    private void enableLogin() {
+        tooManyRequests = false;
+    }
+
     private ArrayList<SomfyTahomaEvent> getEvents() {
```

Comment:
For "loose coupling" it's better to return the `List` interface here instead of the specific `ArrayList` implementation. Then if one day the method returns a list using another implementation the code calling the method doesn't need to be updated.

```suggestion
     private List<SomfyTahomaEvent> getEvents() {
```

Reply:
Fixed. thanks